v2/dtos/deviceservice.go

```diff
@@ -30,9 +30,9 @@ type DeviceService struct {
 // UpdateDeviceService and its properties are defined in the APIv2 specification:
 // https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/UpdateDeviceService
 type UpdateDeviceService struct {
-	Id             *string  `json:"id" validate:"required_without=Name"`
-	Name           *string  `json:"name" validate:"required_without=Id"`
-	BaseAddress    *string  `json:"baseAddress"`
+	Id             *string  `json:"id" validate:"required_without=Name,edgex-update-dto-id"`
+	Name           *string  `json:"name" validate:"required_without=Id,edgex-update-dto-name"`
+	BaseAddress    *string  `json:"baseAddress" validate:"omitempty,gt=0"`
```

Comment:
Why is this needed? what does `gt=0` mean for a `*string` type? Is this way to validate that it isn't an empty string? What about blanks. BaseAddress is a fully qualified URL. Should validate it is a fully qualified URL, it not nil.

Reply:
> Why is this needed? what does gt=0 mean for a *string type? Is this way to validate that it isn't an empty string? What about blanks. 

The gt=0 is used to check string length, but we should check the blank string as you mentioned.

> BaseAddress is a fully qualified URL. Should validate it is a fully qualified URL, it not nil.

Added uri tag for validation. https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-URI_String